src/qilib/configuration_helper/adapters/spi_rack_instrument_adapter.py

```diff
@@ -18,16 +18,19 @@
 OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 """
 from spirack import SPI_rack
+from typing import Optional, Type
 
 from qilib.configuration_helper.instrument_adapter import InstrumentAdapter
 from qilib.utils.python_json_structure import PythonJsonStructure
 
 
 class SpiRackInstrumentAdapter(InstrumentAdapter):
 
-    def __init__(self, address: str) -> None:
+    def __init__(self, address: str, instrument_class: Optional[Type[SPI_rack]] = None) -> None:
         super().__init__(address)
-        self._spi_rack: SPI_rack = SPI_rack(address, baud='115200', timeout=1)
+        if instrument_class is None:
+            instrument_class = SPI_rack
+        self._spi_rack: SPI_rack = instrument_class(address, baud='115200', timeout=1)
```

Comment:
for consistency, shouldn't this be `self._instrument`?

Reply:
SPI_rack is special... it is not a QCoDeS Instrument.... does not inherit from it... it inherits from serial.Serial. I guess that is why the original author of this code wrote it like this?

Reply:
What is your take on it? Leave it as a special snowflake, or change it to an "instrument". We call it an instrument_class and instrument_adapter anyway, right?

Reply:
I say we leave it as is. The adapter defines a property instrument which returns self._spi_rack anyway. your call.

Reply:
I'm fine with it.